pkg/issuer/acme/dns/azuredns/azuredns_test.go

```diff
@@ -130,7 +130,15 @@ func TestGetAuthorizationFederatedSPT(t *testing.T) {
 	// Prepare environment variables adal will rely on. Skip changes for some envs if they are already defined (=live environment)
 	// Envs themselves are described here: https://azure.github.io/azure-workload-identity/docs/installation/mutating-admission-webhook.html
 	if os.Getenv("AZURE_TENANT_ID") == "" {
-		t.Setenv("AZURE_TENANT_ID", "fakeTenantID")
+		// TODO(wallrj): This is a hack. It is a quick way to `DisableInstanceDiscovery` during tests,
+		// to avoid the client attempting to connect to https://login.microsoftonline.com/common/discovery/instance.
+		// It works because there is a special case in azure-sdk-for-go which
+		// disables the instance discovery when the tenant ID is `adfs`. See:
+		// https://github.com/Azure/azure-sdk-for-go/blob/7288bda422654bde520a09034dd755b8f2dd4168/sdk/azidentity/public_client.go#L237-L239
+		// https://learn.microsoft.com/en-us/windows-server/identity/ad-fs/ad-fs-overview
+		//
+		// Find a better way to test this code.
+		t.Setenv("AZURE_TENANT_ID", "adfs")
```

Comment:
Without this change, the tests fail.

> Error:      	Received unexpected error:
>         	            	WorkloadIdentityCredential: unable to resolve an endpoint: server response error:
>         	            	 Get "https://login.microsoftonline.com/common/discovery/instance?api-version=1.1&authorization_endpoint=https%3A%2F%2F127.0.0.1%3A33327%2Ffaketenantid%2Foauth2%2Fv2.0%2Fauthorize": tls: failed to verify certificate: x509: certificate signed by unknown authority
>         	Test:       	TestGetAuthorizationFederatedSPT/token_refresh
>     azuredns_test.go:202: 

You can see an example of the problem in:
 * https://prow.build-infra.jetstack.net/view/gs/jetstack-logs/pr-logs/pull/cert-manager_cert-manager/6679/pull-cert-manager-master-make-test/1752360332161978368

Reply:
The thing about this test is that it was added in #5570 to validate custom token refresh logic for adal (adal doesn't keep track of token expiration, thus the need for a custom refresh function). Now, with the new library in place, we can think of removing the test.

Reply:
The test is indeed a bit outdated, it is now testing the new library instead of our custom refresh logic.
However, I think this test is still important because it tests that we are using the azure library correctly.
We might want to update the tests a bit in the future to focus more on testing the integration itself.
But it does seem like this issue will always happens when we try to test this part of the code (is not specific to the token refresh we are testing here).